tt_metal/impl/buffers/buffer.cpp

```diff
@@ -308,7 +308,7 @@ uint64_t Buffer::core_address(uint32_t core_id) const {
 }
 
 void Buffer::deallocate() {
-    if (this->device_ == nullptr or not this->device_->initialized_) {
+    if (this->device_ == nullptr or not this->device_->initialized_ or this->size_ == 0) {
```

Comment:
How does 0 size buffer even come about?

Reply:
buffer size is set to 0 when it gets deallocated. this is just early exit for buffer that was previously deallocated

Reply:
if it gets deallocated, is it possible to structure the code such that the Buffer object doesn't exist at all?

Reply:
it may be, but that might be outside the scope of this PR since its meant to address a P0 bug